### ISSUE TITLE
545407EE - The Darkness

### ISSUE THREAD

**Issue by AndylgTom** [xenia-canary/game-compatibility#229]
### Validation

- [X] I've read the [FAQ](https://github.com/xenia-canary/xenia-canary/wiki/FAQ).
- [X] The Xenia build used is from a canary branch. (not MLBS/AlexVS/master/pull requests, etc)
- [X] This issue isn't for help (tech support) with Xenia, or a game.
- [X] I've checked if this game doesn't already have a compatibility report.

### Xbox 360 Marketplace link

https://marketplace.xbox.com/en-US/Product/The-Darkness/66acd000-77fe-1000-9115-d802545407ee

### Xenia version

https://github.com/xenia-canary/xenia-canary/commit/eeb96f91a5639c241bdb6d1d5f1dfdc31e8b3e48

### Issues

Strong bloom effect problem, that made everything brighter than it should have been.
By the way, on the ps3 emulator there was the same problem, but it was fixed a long time ago.

*The sound is fine in canary, unlike master.

### Log

[xenia.zip](https://github.com/user-attachments/files/16260581/xenia.zip)

### Screenshots

![545407EE - 2024-07-17T01-08-47](https://github.com/user-attachments/assets/0c457a50-4d51-4beb-a22e-027a099018bf)
![545407EE - 2024-07-17T01-11-56](https://github.com/user-attachments/assets/9bb24830-247d-45fd-abd6-71a8bb1e7a06)
![545407EE - 2024-07-17T01-49-17](https://github.com/user-attachments/assets/c9028838-b03f-429b-a9a9-81bb81f6bdcb)
![545407EE - 2024-07-17T01-49-30](https://github.com/user-attachments/assets/8a4728f6-fdfc-4428-a192-1e1cfc7b7dfd)

Gif difference between original and emulator images ![Animation-1](https://github.com/user-attachments/assets/ec20db40-3dc3-434a-bb82-1429e60bf62f)

### Labels

state-gameplay

**Comment by Margen67** [comment]
Not a valid commit link.

**Comment by Iberra6000** [comment]
The Darkness is fully playable with minimal issues

For anyone wanting to play don't use recommended config cause its got a issue with high fps where ur character runs really fast so cap game at 120fps with vsync on and with that I was able to beat the game with basically no issues  


![Image](https://github.com/user-attachments/assets/10812471-8dda-4b00-bb24-d5a99b29ad75)

**Comment by nfxw** [comment]
Found strange bug.
When you upscale to 1440p - some objects goes green or may be unseen.
But if you upscale to 4k - no rendering issues.

<img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/bdea0f9a-fbe7-44a0-a2c3-95bbd5e48be0" />
<img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/43807aea-50c7-45cc-9bd0-bf21191674b2" />

**Comment by r3538987** [comment]
I had encountered such thing.
Somewhat in middle of the game, there is section with battlefield, which has rain, dark clouds, and some fog. 
In videos this look like this:

<img width="1997" height="1120" alt="Image" src="https://github.com/user-attachments/assets/f46480cb-e092-48bc-95c2-437d70399b65" />

While in game it looked almost inverted. 
(It's same with or without BrightLightFix, FovFix, or on various resolution scales)

<img width="1284" height="719" alt="Image" src="https://github.com/user-attachments/assets/0565d733-0d23-456c-9b68-cc17d6683adc" />

Some sequences have screen going dark, but in my case it went to pure while. 
Had anyone encountered similar behavior? Or maybe even found workarounds? 

When shield ability is unlocked, sometimes effects go crazy.
Sample - https://webmshare.com/play/3gP33

**Comment by calebcharlesnguyen-boop** [comment]
I have a  problem in the beginning of the game where when I start a new game and I look up the screen turns really white. Idk if it frooze because there is no sounds but the ambience.

**Comment by nfxw** [comment]
> I have a problem in the beginning of the game where when I start a new game and I look up the screen turns really white. Idk if it frooze because there is no sounds but the ambience.

If your game freezes, then try to change use_dedicated_xma_thread to false and xma_decoder to "old".

**Comment by Tumatutuma** [comment]
> Found strange bug. When you upscale to 1440p - some objects goes green or may be unseen. But if you upscale to 4k - no rendering issues.


Yep, I'm having the same bug. Too bad since 4k really hits the performance ((

**Comment by goldislead** [comment]
Needs gpu-occlusion-query.

The blown out image is merely auto-exposure that depends on sample counts. The game is rendering threshold passes, reading back 8 visible counts per-frame from OQ, then deriving a scalar.

It also has by-the-numbers ZPD lens flares.

**Comment by Psyko66** [comment]
Hey! @r3538987 I tested the game in Xenia Edge and the fog in that section of the battlefield looks as it should (with fixes applied) but the game runs worse and it has more effects going crazy, so i think that is something related to Xenia Canary, maybe some configuration but I haven't figured out what is it

**Comment by goldislead** [comment]
> Hey! [@r3538987](https://github.com/r3538987) I tested the game in Xenia Edge and the fog in that section of the battlefield looks as it should (with fixes applied) but the game runs worse and it has more effects going crazy, so i think that is something related to Xenia Canary, maybe some configuration but I haven't figured out what is it

Don't use any "fixes". That specular brightness patch is a hack workaround, not a fix.

If you want to test it as a one-for-one against the latest Canary, [use the open PR](https://github.com/xenia-canary/xenia-canary/pull/1016) build that adds occlusion query support to Canary, not Edge.

**Comment by Psyko66** [comment]
> 
> Don't use any "fixes". That specular brightness patch is a hack workaround, not a fix.
> 
> If you want to test it as a one-for-one against the latest Canary, [use the open PR](https://github.com/xenia-canary/xenia-canary/pull/1016) build that adds occlusion query support to Canary, not Edge.

It still looks bad in that section even in the latest Canary with and without the bloom patch and occlusion query modified

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/401b7d77-d3e4-4b38-bf1d-75c9aa80734f" /> Without patch

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/ef8c25fa-aa85-4bb2-bb58-086865a7f280" /> With patch

Both pictures with occlusion query set to -1

**Comment by goldislead** [comment]
You're not doing what I said. Join the Xenia discord and seek assistance there if you need it. This isn't really the place for that.

**Comment by BorealisEcho** [comment]
As of fbd620c_canary_experimental (2026-05-30) release, this game benefits from ZPD occlusion queries.

~~Set `occlusion_query = "fast-alt"` or `occlusion_query = "strict"` to fix exposure.~~

As of 3d233a5_canary_experimental, using `occlusion_query = "fast"` now works also.

**Comment by nfxw** [comment]
The game is no longer required patches.
Both issue with lightning and fps are fixed on dd858e2.

<img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/401e2750-4647-4b0a-adbf-921cd77fa3e2" />

**Comment by n3xu589** [comment]
> Found strange bug.
> When you upscale to 1440p - some objects goes green or may be unseen.
> But if you upscale to 4k - no rendering issues.
> 
> <img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/bdea0f9a-fbe7-44a0-a2c3-95bbd5e48be0" />
> <img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/43807aea-50c7-45cc-9bd0-bf21191674b2" />

This bug still exists.

**Comment by goldislead** [comment]
It's not a bug, just issues inherent to resolution scaling.
Countless shaders in this game assume fixed UV offsets, far beyond just post-process effects.
The solution lies in some very belabored patches, not Canary's GPU code.
